src/engine/Server.cpp

```diff
@@ -915,17 +960,22 @@ bool Server::checkAccessToken(
 // _____________________________________________________________________________
 
 std::optional<std::string_view> Server::checkParameter(
-    const ad_utility::HashMap<std::string, std::string>& parameters,
+    const ad_utility::url_parser::ParamValueMap& parameters,
     std::string_view key, std::optional<std::string_view> value,
     bool accessAllowed) {
   if (!parameters.contains(key)) {
     return std::nullopt;
   }
+  if (parameters.at(key).size() > 1) {
+    return std::nullopt;
```

Comment:
Do we want to throw here?  silently ignoring seems a little bit confusing towards the users for me.

Reply:
Resolved by using `getParameterCheckAtMostOnce`